### PR TITLE
Improve location using GeoLite2

### PR DESCRIPTION
**Pull request by Neylix**
# Description

Geo localisation is implemented with ip2location database, but I figured that this db wasn't really accurate for several IPs and often return main ISP router location than user router location.
Maxmind provide a free location DB (GeoLite2) which is more accurate than ip2location, so I propose to use this one instead

## Type of change

- Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Tested with several IP in France and India, location are way better.

# Checklist:

- My code follows the style guidelines of this project
- I have performed a self-review of my own code
- I have commented my code, particularly in hard-to-understand areas
- I have made corresponding changes to the documentation
- My changes generate no new warnings
- I have added tests that prove my fix is effective or that my feature works
- New and existing unit tests pass locally with my changes
- Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: config/config.exs [hunk 1]
```diff
@@ -136,6 +136,15 @@ config :archethic, Archethic.Networking.IPLookup.NATDiscovery,
 config :archethic, Archethic.Networking.IPLookup.RemoteDiscovery,
   provider: Archethic.Networking.IPLookup.RemoteDiscovery.IPIFY
 
+config :geolix,
+  databases: [
+    %{
+      id: :city,
+      adapter: Geolix.Adapter.MMDB2,
```
Comment (apoorv-2204) [inline config/config.exs:143]: This is really nice, I was not able to find the mamind db . Sam will be really happy : ) nice

Comment (apoorv-2204) [reply config/config.exs:143]: We will now have very less offset . We can now think about Network patches ?

Comment (samuel-uniris) [reply config/config.exs:143]: While it might improve location, network patch is something which might improve P2P calls, but not really priority for now.

**Comment by Neylix** [comment]
> https://github.com/elixir-geolix/adapter_mmdb2 might be lighter choice so we could add an impl for MaxMind instead of Ip2location without changing too much configuration, and supervision tree, ...
> 
> As we can call the function directly from the lib.
> 
> Otherwise we could embrace Geolix as abstraction but might tight-coupled this part of the code to this library

Maybe directly use https://hexdocs.pm/mmdb2_decoder/MMDB2Decoder.html
Instead of using Geolix

**Comment by samuel-uniris** [comment]
I think you might use directly the GenServer, ETS is great when you are dealing with high read concurrency but here I don't think you will face bottlenecks with a plain GenServer .

Before we used ETS because the way

File: mix.exs [hunk 2]
```diff
@@ -75,6 +75,7 @@ defmodule Archethic.MixProject do
 
       # P2P
       {:ranch, "~> 2.1", override: true},
+      {:mmdb2_decoder, "~> 3.0"},
```
Comment (apoorv-2204) [inline mix.exs:78]: Is it better than maxmind geo database?

Comment (Neylix) [reply mix.exs:78]: This is the same database, but just use a lighter library to fetch the data.
Geolix and GeolixAdapterMmdb2 were to eavy and finaly use Mmdb2Decoder

**Review by samuel-uniris — APPROVED** [review]

**Pull request merged**